custom_components/tuya_local/devices/breville_smartairconnect_purifier.yaml

```diff
@@ -0,0 +1,125 @@
+name: Breville Smart Air Connect Purifier
+products:
+  - id: tltdxtehttjbkjni
+primary_entity:
+  entity: fan
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: true
+          icon: "mdi:air-purifier"
+        - dps_val: false
+          icon: "mdi:air-purifier-off"
+    - id: 3
+      name: mode
+      type: string
+      mapping:
+        - dps_val: "manual"
+          value: "Manual"
+        - dps_val: "auto"
+          value: "Auto"
+    - id: 4
+      type: string
+      name: speed
+      mapping:
+        - dps_val: low
+          value: 33
+        - dps_val: mid
+          value: 66
+        - dps_val: high
+          value: 100
+secondary_entities:
+  - entity: sensor
+    name: PM2.5
+    icon: "mdi:air-filter"
+    class: pm25
+    dps:
+      - id: 2
+        name: sensor
+        class: measurement
+        type: integer
+        unit: µg/m3
+  - entity: select
```

Comment:
This is duplicating the mode under fan.  A separate entity for it is not required.

Reply:
When it's placed under the primary entity, it seems to be informational only and doesn't allow the user to change the value - but does show the current setting on the main Lovelace panel for the device (see below). Having it as a sub entity also allows for configuration of the option. I'm more than happy to deduplicate it though. Do you have a preference on where the dp should be set?
![smart_air_connect_panel](https://user-images.githubusercontent.com/6414258/221334296-e65bb089-0bf0-4467-a7fd-d927039c72d0.jpg)

Reply:
Ah, I missed that.  Under fan, the name should be preset_mode, not mode.

Reply:
Ah, cool, I understand how that element works now. I've updated the name to preset_mode and removed the secondary entity.